packages/generator/src/generator-utils.ts

```diff
@@ -190,7 +190,11 @@ export function getGenericParameters(
 ): string {
   const params = [entityClassName];
   if (prop.isEnum) {
-    params.push("'Edm.Enum'");
+    if (prop.isCollection) {
+      params.push(`typeof ${prop.jsType}`);
+    } else {
+      params.push(`${prop.jsType}`);
+    }
```

Comment:
This is hard to grasp and is is easy to make errors here, maybe we should rather write this down explicitly as we have done it in some other cases. It is a bit more verbose, but so much easier to read.

```ts
if (isCollection) {
  if(isEnum) {
    ...
  } else if (isComplex) {
    ...
  } else {
    ...
 }
} else {
    if(isEnum) {
    ...
  } else if (isComplex) {
    ...
  } else {
    ...
 }
}
```